src/Roslyn.Suppressions/Roslyn.Suppressions.UnitTests/SuppressionCheckerTests.cs

```diff
@@ -187,20 +187,19 @@ public void IsMatch_RuleIdsAreDifferent_ReturnsFalse()
         private const string DiagFileText = "0\n1\n2\n3 text \n\n";
         private const string DiagSelectedText = "text";
         private const int DiagRoslynLineNumber = 3; // the 0-base line in which the word "text" appears
-        private const int MatchingSonarLineNumber = DiagRoslynLineNumber + 1;
 
         [TestMethod]
-        [DataRow(DiagFileName, DiagRuleId, MatchingSonarLineNumber, DiagHash, true)]
-        [DataRow("c:\\wrong file name.cs", DiagRuleId, MatchingSonarLineNumber, DiagHash, false)]
-        [DataRow(DiagFileName, "wrong rule id", MatchingSonarLineNumber, DiagHash, false)]
+        [DataRow(DiagFileName, DiagRuleId, DiagRoslynLineNumber, DiagHash, true)]
```

Comment:
The suppression checker is now matching Roslyn `Diagnostic` instances against the new `SuppressedIssue` type.
The line numbers have already been translated from 1-based to 0-based so the comparisons are simpler.